examples/alicebob-grant-and-line-by-line-PRE.py

```diff
@@ -78,16 +56,10 @@ def enact_policy(self, ursula, hrac, payload):
         print("PREs per second: {}".format(counter / seconds))
         print("********************************")
 
-    ciphertext, capsule = pre.encrypt(ALICE.public_key(EncryptingPower), plaintext)
-
-    message_kit = MessageKit(ciphertext=ciphertext, capsule=capsule,
-                      alice_pubkey=ALICE.public_key(EncryptingPower))
+    data_source = DataSource(policy_pubkey_enc=policy.public_key())
+    message_kit, _signature = data_source.encapsulate_single_message(plaintext)
```

Comment:
This change manages to convey the greater flexibility afforded to developers in using a DataSource character to handle encryption on behalf of Alice, and yet is easier to understand than the code it replaces. I tip my hat to you sir. 

I will still write some instructions (JN style) plus an in-line note explaining this in more detail here, since it would be cool if they totally get everything in this file while reading it – then they can go dig through all the lower-level stuff.

Reply:
Sure.  And then, in #246, I have evolved it a state further.